core_data_modules/data_models/scheme.py

```diff
@@ -34,40 +38,43 @@ def get_code_with_control_code(self, control_code):
 
     def get_code_with_match_value(self, match_value):
         for code in self.codes:
-            if match_value in code.match_values:
+            if code.match_values is not None and match_value in code.match_values:
```

Comment:
I would prefer to move an assert that match_value was not none to a precondition after the first line - it doesn't seem to make sense to me to call this method with match_value == None

Reply:
Oh, sorry, yeah, it's inside the for loop, ignore.